plugins/inputs/net/net.go

```diff
@@ -129,6 +133,26 @@ func (n *NetIOStats) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
+// Get the interface speed from /sys/class/net/*/speed file. returns -1 if unsupported
+func getInterfaceSpeed(ioName string) int64 {
+	sysPath := os.Getenv("HOST_SYS")
+	if sysPath == "" {
+		sysPath = "/sys"
+	}
+
+	raw, err := os.ReadFile(filepath.Join(sysPath, "class", "net", ioName, "speed"))
+	if err != nil {
+		return -1
+	}
+	speedStr := string(raw)
+
+	speed, err := strconv.ParseInt(speedStr, 10, 64)
```

Comment:
The files actually do have new lines on Linux, so:

```suggestion
	speed, err := strconv.ParseInt(strings.TrimSuffix(string(raw), "\n"), 10, 64)
```

Reply:
Probably should update the test case files as well.